ext/prism/extension.c

```diff
@@ -42,17 +42,11 @@ ID rb_id_source_for;
 /******************************************************************************/
 
 /**
- * Check if the given VALUE is a string. If it's nil, then return NULL. If it's
- * not a string, then raise a type error. Otherwise return the VALUE as a C
- * string.
+ * Check if the given VALUE is a string. If it's not a string, then raise a
+ * TypeError. Otherwise return the VALUE as a C string.
  */
 static const char *
 check_string(VALUE value) {
-    // If the value is nil, then we don't need to do anything.
-    if (NIL_P(value)) {
```

Comment:
Just curious why this line is going away?

Reply:
Because all callers are already checking for nil before, and returning NULL is dangerous e.g. if later strlen() or so is called on it.
It's explained in the commit message of https://github.com/ruby/prism/pull/3077/commits/8197be883e2d8f528d1ebf1821e201f45ee7fbaa